src/attribute.c

```diff
@@ -183,3 +183,164 @@ u8 sub_08017CE0(u16 r5)
     }
     return r6;
 }
+
+void sub_08017D84(u8 r0, u16 r1)
+{
+    struct SaveBlock1 *sb1 = gSaveBlock1Ptr;
+    if (r1 > gUnk_08B80178[9])
```

Comment:
Are you sure `gUnk_08B80178` is not a struct?

Reply:
I try listing xref to the operand, and I get something like this:
` LSLS            R0, R0, #0x18
   LDR             R1, =gUnk_08B80178
   LSRS            R0, R0, #0x17
  ADDS            R0, R0, R1
   LDRH            R0, [R0]
   BX              LR`